assets/model_monitoring/components/src/feature_importance_metrics/compute_feature_importance.py

```diff
@@ -220,9 +233,70 @@ def write_to_mltable(explanations, dataset, file_path, categorical_features):
                                          constants.METRIC_NAME_COLUMN,
                                          constants.FEATURE_CATEGORY_COLUMN,
                                          constants.THRESHOLD_VALUE], index=[0])
-    print(explanation_data)
     metrics_data = pd.concat(explanation_data)
-    spark_data = convert_pandas_to_spark(metrics_data)
+    return metrics_data
+
+
+def create_local_feature_importance_df(explanations, dataset, task_type):
+    """Write local feature importance values to mltable.
+
+    :param explanations: list of local feature importances in the order of the baseline columns
+    :type explanations: list[float]
+    :param dataset: dataset to derive feature names
+    :type dataset: pandas.Dataframe
+    :param task_type: the task type (regression or classification) of the resulting model
+    :type task_type: string
+    :return: local feature importance sample table
+    :rtype: pandas.Dataframe
+    """
+    # Create column names
+    columns = [constants.ROW_INDEX]
+    if task_type == constants.CLASSIFICATION:
+        columns.append(constants.CLASS)
+    feature_names = list(dataset.columns)
+    columns += feature_names
+
+    sample_data = pd.DataFrame(columns=columns)
+
+    # Populate importances
+    for index in range(len(explanations)):
+        new_row = {}
+        new_row[constants.ROW_INDEX] = index
+
+        if task_type == constants.CLASSIFICATION:
+            for potential_class in range(len(explanations[index])):
+                # TODO: Class is currently the index of the class in the local explanations matrix,
+                # update to be mapped to class name
+                new_row[constants.CLASS] = potential_class
+                local_explanation = explanations[index][potential_class][0]
```

Comment:
Could you please show an example of this `explanations[index][potential_class]` object? Better if you can use the input data in a unit test so that later developers can easily pick up.

Reply:
From the design doc you shared, I can see there is always a list of one dictionary object of local explanation. Curious if you know what that means.

Reply:
For classification, the local explanation has a row for each class. Each row can be interpreted as the feature importance values given class X is predicted. Does this answer your question?